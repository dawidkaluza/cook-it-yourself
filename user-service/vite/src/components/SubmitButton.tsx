import React from "react";

export const SubmitButton = (props: {
  loading?: boolean,
  style?: string
  children: React.ReactNode;
}) => {
  return (
    <button
      type="submit"
      disabled={props.loading}
      className={
        "bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-6 rounded disabled:opacity-75 disabled:hover:bg-blue-500 "
        + (props.style ?? "")
      }
    >
      {props.children}
    </button>
  );
};