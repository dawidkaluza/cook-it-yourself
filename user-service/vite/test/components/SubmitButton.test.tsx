import {SubmitButton} from "../../src/components/SubmitButton.tsx";
import {render, screen} from "@testing-library/react";

describe("SubmitButton component", () => {
  test("render with required props", () => {
    // Given
    // When
    const component = render(<SubmitButton>Submit form</SubmitButton>);

    // Then
    const btnEl = screen.getByRole("button", { name: "Submit form" });
    expect(btnEl).toBeDefined();
    expect(btnEl.getAttribute("disabled")).toBeNull();

    expect(component.container).toMatchSnapshot();
  });

  test("render with all props", () => {
    // Given
    // When
    const component = render(<SubmitButton loading={true} style="w-full">Submit form</SubmitButton>);

    // Then
    const btnEl = screen.getByRole("button", { name: "Submit form" });
    expect(btnEl).toBeDefined();
    expect(btnEl.getAttribute("class")).toContain("w-full");
    expect(btnEl.getAttribute("disabled")).not.toBeNull();

    expect(component.container).toMatchSnapshot();
  })
});