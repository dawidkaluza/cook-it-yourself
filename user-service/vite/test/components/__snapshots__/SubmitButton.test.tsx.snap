// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubmitButton component > render with all props 1`] = `
<div>
  <button
    class="bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-6 rounded disabled:opacity-75 disabled:hover:bg-blue-500 w-full"
    disabled=""
    type="submit"
  >
    Submit form
  </button>
</div>
`;

exports[`SubmitButton component > render with required props 1`] = `
<div>
  <button
    class="bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-6 rounded disabled:opacity-75 disabled:hover:bg-blue-500 "
    type="submit"
  >
    Submit form
  </button>
</div>
`;
